Tangem/App/ViewModels/WalletModel.swift

```diff
@@ -119,216 +212,140 @@ class WalletModel: ObservableObject, Identifiable {
             .delay(for: 0.3, scheduler: DispatchQueue.main)
             .dropFirst()
             .receive(on: updateQueue)
-            .setFailureType(to: Error.self)
-            .flatMap { [weak self] _ in
-                self?.loadRates() ?? .justWithError(output: [:])
+            .receiveValue { [weak self] _ in
+                self?.loadRates()
             }
+            .store(in: &bag)
+
+        walletManager.statePublisher
+            .filter { !$0.isInitialState }
+            .combineLatest(walletManager.walletPublisher) // listen pending tx
             .receive(on: updateQueue)
-            .receiveValue { [weak self] in self?.updateRatesIfNeeded($0) }
+            .sink { [weak self] newState, _ in
+                self?.walletManagerDidUpdate(newState)
+            }
             .store(in: &bag)
 
-        $state.dropFirst()
-            .combineLatest($rates.dropFirst())
-            .map { $0.0 } // Move on latest value state
-            .delay(for: 0.3, scheduler: DispatchQueue.main)
-            .receiveValue { [weak self] value in
-                self?.walletDidChange.send(value)
+        ratesRepository
+            .ratesPublisher
+            .compactMap { [tokenItem] rates -> Decimal? in
+                guard let currencyId = tokenItem.currencyId else { return nil }
+
+                return rates[currencyId]
+            }
+            .removeDuplicates()
+            .sink { [weak self] rate in
+                guard let self else { return }
+
+                AppLog.shared.debug("🔄 Rate updated for \(name)")
+                _rate.send(rate)
             }
             .store(in: &bag)
+
+        _state
+            .combineLatest(_rate)
```

Comment:
```suggestion
        _state.dropFirst()
            .combineLatest(_rate.dropFirst())
```
я бы добавил `.dropFirst()`, а то при подписке сразу будет currentValue = .created, раньше по такой логике работало.